mole/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+            encrypted_device_info: format!("\\x{}", hex::encode(encrypted_device_info)),
+        };
+        post_blocking::<InsertChannelMonitorOne>(&client, &self.backend_url, variables)?;
+
+        Ok(())
+    }
+
+    pub fn get_channel_monitor_ids(&self) -> Result<Vec<String>> {
+        let token = self.auth.query_token()?;
+        let client = build_client(Some(&token))?;
+        let variables = get_channel_monitor_channel_ids::Variables {};
+        let data =
+            post_blocking::<GetChannelMonitorChannelIds>(&client, &self.backend_url, variables)?;
+        let list = data
+            .channel_monitor
+            .into_iter()
+            .map(|cm| cm.channel_id.replace("\\x", ""))
+            .collect();
+
+        Ok(list)
+    }
+
+    pub fn read_channel_monitor(&self, channel_id: &str) -> Result<Vec<u8>> {
+        let token = self.auth.query_token()?;
+        let client = build_client(Some(&token))?;
+        let variables = get_latest_channel_monitor::Variables {
+            channel_id: format!("\\x{channel_id}"),
+        };
+        let data = post_blocking::<GetLatestChannelMonitor>(&client, &self.backend_url, variables)?;
+        let binary = hex::decode(
+            data.channel_monitor[0]
+                .encrypted_channel_monitor
+                .replace("\\x", ""),
+        )
+        .map_to_runtime_error(
+            GraphQlRuntimeErrorCode::GenericError,
+            "Could not decode hex encoded binary",
+        )?;
+
+        Ok(binary)
+    }
+
+    pub fn write_channel_manager(&self, encrypted_channel_manager: &Vec<u8>) -> Result<()> {
+        let token = self.auth.query_token().unwrap();
+        let client = build_client(Some(&token))?;
+        let variables = insert_channel_manager_one::Variables {
+            encrypted_channel_manager: format!("\\x{}", hex::encode(encrypted_channel_manager)),
+        };
+        post_blocking::<InsertChannelManagerOne>(&client, &self.backend_url, variables)?;
+
+        Ok(())
+    }
+
+    pub fn read_channel_manager(&self) -> Result<Vec<u8>> {
+        let token = self.auth.query_token()?;
+        let client = build_client(Some(&token))?;
+        let variables = get_latest_channel_manager::Variables {};
+        let data = post_blocking::<GetLatestChannelManager>(&client, &self.backend_url, variables)?;
+        let binary = hex::decode(
+            data.channel_manager[0]
+                .encrypted_channel_manager
+                .replace("\\x", ""),
+        )
+        .map_to_runtime_error(
+            GraphQlRuntimeErrorCode::GenericError,
+            "Could not decode hex encoded binary",
+        )?;
+
+        Ok(binary)
+    }
+}
+
+fn error_is_data_structure_related(
+    error: &graphql::perro::Error<graphql::GraphQlRuntimeErrorCode>,
+) -> bool {
+    matches!(
+        error,
+        Error::RuntimeError {
+            code: GraphQlRuntimeErrorCode::GenericError,
+            ..
+        }
+    )
```

Comment:
Is a `GraphQlRuntimeErrorCode::GenericError` always indicative of a successful request without results? Even if that's the case at the moment, we could in the future use if for other scenarios, which could break this logic here.

Reply:
Looks like we're getting rid of that code altogether ..

Reply:
e1cdab62baf5965321dad228b8dae5ac7a9a02a2